AsynchronouslyLoadingImagesIntoTableAndCollectionViews/Async Image Loading/ImageURLProtocol.swift

```diff
@@ -30,12 +31,15 @@ class ImageURLProtocol: URLProtocol {
             return
         }
 
-
-        block = DispatchWorkItem(block: {
-            if self.cancelledOrComplete == false {
+        self.asyncTask = Task { [weak self] in
```

Comment:
@mattmassicotte When I turned on "complete" this causes an error with the message of 
```
Passing closure as a 'sending' parameter risks causing data races between code in the current task and concurrent execution of the closure; this is an error in the Swift 6 language mode
```

I assume that self could be written by multiple requests to `startLoading()` so the system will not let me do this. I'll think about how to approach this because I'm not writing to `self` (afaik) but only using `self` as an argument when letting the `urlClient` that I have received a response and did load.  

🤔